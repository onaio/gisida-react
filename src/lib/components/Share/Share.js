import React from 'react';
import { useToasts } from 'react-toast-notifications';
/**
 * Component handles sharing site instances by copying parmeterized url to clipboard
 */
const Share = () => {
    const { addToast } = useToasts();
    const onCopyToClipBoard = () => {
        /**
         * Unfortunately there is no standard API for clipboard operations,
         * so we're left with the hacky way of using a HTML input element to fit our needs
         */
        const el = document.createElement('input');
        el.value = window.location.href;
        el.id = "url-input";
        document.body.appendChild(el);
        el.select();
        document.execCommand("copy");
        el.remove();
        addToast('Link copied to clipboard', { appearance: 'success', autoDismiss: true, autoDismissTimeout: 2000 });
    };

    return (<div>
        <li>
            <a href="#" onClick={() => onCopyToClipBoard()}>
                <span className="glyphicon glyphicon-share" />
            </a>
        </li>
    </div>);
}


export default Share;