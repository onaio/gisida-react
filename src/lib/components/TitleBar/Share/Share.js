import React from 'react';
/**
 * Component handles sharing site instances by copying parmeterized url to clipboard
 */
const Share = () => {
  const onCopyToClipBoard = () => {
    /**
     * Unfortunately there is no standard API for clipboard operations,
     * so we're left with the hacky way of using a HTML input element to fit our needs
     */
    const el = document.createElement('input');
    el.value = window.location.href;
    el.id = 'url-input';
    document.body.appendChild(el);
    el.select();
    document.execCommand('copy');
    el.remove();
    console.log('Lets make a toast');
    alert('Linked copied');
  };

  return (
    <div>
      <li>
        <a href="#" title='Share' alt='Share'  onClick={() => onCopyToClipBoard()}>
          <span className="glyphicon glyphicon-share" />
        </a>
      </li>
    </div>
  );
};

export default Share;
