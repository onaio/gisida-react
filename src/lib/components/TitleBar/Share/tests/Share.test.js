import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Share from '../Share';

describe('Share', () => {
  beforeAll(() => {
    Object.defineProperty(document, 'execCommand', {
      copy: jest.fn().mockImplementation(() => {
        return '';
      }),
    });
  });
  it('renders correctly', () => {
    const wrapper = shallow(<Share />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
