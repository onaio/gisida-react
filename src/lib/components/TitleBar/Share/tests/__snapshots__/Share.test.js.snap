// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share renders correctly 1`] = `
<div>
  <li>
    <a
      alt="Share"
      href="#"
      onClick={[Function]}
      title="Share"
    >
      <span
        className="glyphicon glyphicon-share"
      />
    </a>
  </li>
</div>
`;
